### ISSUE TITLE
Problem with etl::vector::emplace_back

### ISSUE THREAD

**Issue by r2r0** [ETLCPP/etl#290]
Code below does not compile, where similar code based on std::vector compiles (and works) without any problem.

```
struct SomeObject {};
etl::vector<SomeObject*, 10> vec {};
vec.emplace_back(nullptr);
```

To avoid problem I have to define simple wrapper i.e.
```
struct SomeObjectWrapper
{
    SomeObjectWrapper(SomeObject* a_ptr) : ptr(a_ptr) {}
    SomeObject* ptr;
};
etl::vector<SomeObjectWrapper, 10> vec {};
```

Is it intentional behavior?

Compiler: GCC 9.3.1 on OpenSUSE

**Comment by jwellbelove** [comment]
No, it's not intentional, I've just never tested for that scenario.
I'll take a look to see what can be done.

**Comment by jwellbelove** [comment]
It seems that I neglected to give the `etl::vector` specialisation for pointers an `emplace_back` implementation.
I'll raise an issue for this.

**Comment by jwellbelove** [comment]
Can you use push_back for the moment?

**Comment by r2r0** [comment]
Thanks for promp follow up.
No problem, for such basic type push_back works well.

**Comment by jwellbelove** [comment]
Fixed in release 18.17.2
